apis/v1alpha1/authenticationfilter_types.go

```diff
@@ -0,0 +1,167 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:categories=nginx-gateway-fabric,shortName=authfilter;authenticationfilter
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+
+// AuthenticationFilter configures request authentication and is
+// referenced by HTTPRoute and GRPCRoute filters using ExtensionRef.
+type AuthenticationFilter struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the desired state of the AuthenticationFilter.
+	Spec AuthenticationFilterSpec `json:"spec"`
+
+	// Status defines the state of the AuthenticationFilter, following the same
+	// pattern as SnippetsFilter: per-controller conditions with an Accepted condition.
+	//
+	// +optional
+	Status AuthenticationFilterStatus `json:"status"`
+}
+
+// +kubebuilder:object:root=true
+
+// AuthenticationFilterList contains a list of AuthenticationFilter resources.
+type AuthenticationFilterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []AuthenticationFilter `json:"items"`
+}
+
+// AuthenticationFilterSpec defines the desired configuration.
+// +kubebuilder:validation:XValidation:message="for type=Basic, spec.basic must be set",rule="self.type == 'Basic' ? self.basic != null : true"
```

Comment:
Since Basic is optional, no need for this validation.

Reply:
Yeah, let's require `type`. I think this validation is fine then. Could it be simplified to

```
rule="!(!has(self.basic) && self.type == 'Basic')"
```